hooks/src/index.js

```diff
@@ -184,7 +184,23 @@ export function useReducer(reducer, initialState, init) {
 
 		if (!currentComponent._hasScuFromHooks) {
 			currentComponent._hasScuFromHooks = true;
-			const prevScu = currentComponent.shouldComponentUpdate;
+			let prevScu = currentComponent.shouldComponentUpdate;
+			const prevCWU = currentComponent.componentWillUpdate;
+
+			// If we're dealing with a forced update `shouldComponentUpdate` will
+			// not be called. But we use that to update the hook values, so we
+			// need to call it.
+			currentComponent.componentWillUpdate = function(p, s, c) {
+				if (this._force) {
+					let tmp = prevScu;
+					// Clear to avoid other sCU hooks from being called
+					prevScu = undefined;
```

Comment:
We're:
- defining an instance var
- clearing the instance var
- calling a method which should have access to the instance var (does it?)
- restoring the instance var

I'm reviewing on my phone so I might be missing some of the flow - but I'm missing how the `prevScu = undefined` is having an impact. Did we mean to do `currentComponent.shouldComponentUpdate = undefined`?

Reply:
The gist of why the clearing is needed is that we kinda "misuse" `shouldComponentUpdate` to update the hook values, whereas that method is typically expected to be pure. In the patched `shouldComponentUpdate` method in the hooks addon we call the previous `shouldComponentUpdate` method in case the component already had one. The signals addon for example does this.

Now here's the problem: In a forced update we _still_ want to update all hook values, but not call `shouldComponentUpdate`. Doing so would be a violation of the API contract. To work around this temporarily unsetting `prevScu ` ensures that we only call the hooks overwritten method for `shouldComponentUpdate` and not the other ones.

It's a little weird for sure and something we're planning to make cleaner in v11.

Reply:
Fair - but I still don't understand how `prevScu = undefined` is doing anything? It's an instance variable, which shouldn't be in play when `this.shouldComponentUpdate(p, s, c);` is invoked?

Reply:
NVM, I see it now!